hail/python/hail/expr/functions.py

```diff
@@ -3905,7 +3905,7 @@ def group_by(f: Callable, collection) -> DictExpression:
 @typecheck(f=func_spec(2, expr_any),
            zero=expr_any,
            collection=expr_oneof(expr_set(), expr_array()))
-def fold(f: Callable, zero, collection) -> Expression:
```

Comment:
drive-by fix. Hinting the superclass of the returned value causes lint issues (in my IDE at least), leaving it out and using duck typing doesn't. Can remove if you want, but I've been stripping these out when I see them.

Reply:
Hmm. Pyright complains about our use of `Callable`, but I don't understand the error message [1]. With or without this change, pyright says the type of
```
fold(lambda _: 3, literal(0), literal([1]))
```
is `Expression`. What type do you get?


Hmm, doesn't seem like there's integration for Pyright with IntelliJ unfortunately. The Python community seems to unifying on it since it's remarkably good.

[1]: `Illegal type annotation: variable not allowed unless it is a type alias (lsp)`

Reply:
Ohhhh, I bet this is because your IDE thinks the return type is the return type of the function after modification by `@typecheck`